src/ExpressionEvaluator/Core/Source/ResultProvider/Helpers/DynamicHelpers.cs

```diff
@@ -0,0 +1,71 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using Microsoft.VisualStudio.Debugger.Metadata;
+
+namespace Microsoft.CodeAnalysis.ExpressionEvaluator
+{
+    internal static class DynamicHelpers
+    {
+        private static readonly bool[] TrueArray = new[] { true };
+
+        public static bool[] GetDynamicFlags(this IList<CustomAttributeData> attributes)
+        {
+            foreach (var attribute in attributes)
+            {
+                if (attribute.Constructor.DeclaringType.IsType("System.Runtime.CompilerServices", "DynamicAttribute"))
+                {
+                    var arguments = attribute.ConstructorArguments;
+                    if (arguments.Count == 0)
+                    {
+                        return TrueArray;
+                    }
+                    else if (arguments.Count == 1)
+                    {
+                        var argumentType = arguments[0].ArgumentType;
+                        if (argumentType.IsArray && argumentType.GetElementType().IsBoolean())
+                        {
+                            // Per https://msdn.microsoft.com/en-us/library/system.reflection.customattributetypedargument.argumenttype(v=vs.110).aspx,
+                            // if ArgumentType indicates an array, then Value will actually be a ReadOnlyCollection.
+                            var collection = (ReadOnlyCollection<CustomAttributeTypedArgument>)arguments[0].Value;
+                            var builder = ArrayBuilder<bool>.GetInstance(collection.Count);
+                            foreach (var arg in collection)
+                            {
+                                builder.Add((bool)arg.Value);
+                            }
+                            return builder.ToArrayAndFree();
+                        }
+                    }
+                }
+            }
+
+            return null;
+        }
+
+        public static T[] Skip<T>(this T[] oldArray, int count)
+        {
```

Comment:
This is only used for skipping 1 element of bool[]. Consider changing to: static bool[] SkipOne(this bool[] oldArray).